spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/lock/DefaultLockRepository.java

```diff
@@ -157,8 +160,8 @@ public void delete(String lock) {
 	@Transactional(isolation = Isolation.SERIALIZABLE, timeout = 1)
 	@Override
 	public boolean acquire(String lock) {
-		deleteExpired(lock);
-		if (this.template.update(this.updateQuery, new Date(), this.region, lock, this.id) > 0) {
+		if (this.template.update(this.updateQuery, new Date(), this.region, lock, this.id,
```

Comment:
Wll, I think without `deleteExpired()` over here we are missing the fact that another node in the cluster won't be able to acquire the lock since it was removed. We are not able to reacquire it with the current node because it is expired, but the lock still is not available for another clients to insert because the row for an old is still there.

Any ideas how to deal with that, please, with your new approach?

Reply:
> another node in the cluster won't be able to acquire the lock since it was removed
This point is covered by `JdbcLockRegistryDifferentClientTests.testOutOfDateLockTaken()`

> reacquire it with the current node because it is expired
I wrote `JdbcLockRegistryTests.testReentrantLockAfterExpiration()` to cover this case